tests/org.eclipse.n4js.ide.tests/src/org/eclipse/n4js/ide/tests/contentassist/CompletionWithImportsTest.xtend

```diff
@@ -0,0 +1,262 @@
+/**
+ * Copyright (c) 2020 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ * Contributors:
+ * NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.ide.tests.contentassist;
+
+import java.util.List
+import org.eclipse.n4js.ide.tests.server.AbstractCompletionTest
+import org.junit.Test
+
+/**
+ * Code completion tests for scenarios that also might add an import statement 
+ */
+public class CompletionWithImportsTest extends AbstractCompletionTest {
+
+	/** Some default modules that export a number of classes for all tests. */
+	override final List<Pair<String, String>> getDefaultTestModules() {
+		return #[
+			"MA"  -> '''
+						export class A1 {}
+						export class A2 {}''',
+			"MBA" -> '''
+						export class B1 {}
+						export class A2 {}'''];
+	}
+
+	@Test
+	def void test01() {
+		test('''
+			let x = new A1<|>
+		''', ''' 
+			(A1, Class, MA, , , 00000, , , , ([0:12 - 0:14], A1), [([0:0 - 0:0], import {A1} from "MA";
+			)], [], , )
```

Comment:
What about applying the changes sent by the server and then using the resulting changed code for comparison (to make the test more readable)?
If `TextEdit`s are sent, here, then there is already method `AbstractIdeTest#applyTextEdits()` that might help.

Reply:
Thanks, good idea, but I also want to validate the label, icon, description besides the actual replacement string.